### ISSUE TITLE
what is the order when I call tf.keras.Layers.BatchNormation.get_weights() ? my tensorflow version is 2.2.0?

### ISSUE THREAD

**Issue by daeing** [tensorflow/tensorflow#55444]
<em>Please make sure that this is an issue related to keras.
tag:keras_template</em>

**Important Notice**

Please note that `tf.keras` code was moved entirely to
[keras-team/keras](https://github.com/keras-team/keras) repository

You can open any code/doc bugs, performance issues, and feature requests
 in [keras-team/keras](https://github.com/keras-team/keras/issues) repository

`tf.keras` related issues opened in
[tensorflow/tensorflow](https://github.com/tensorflow/tensorflow) repository may
not get attention as [keras-team/keras](https://github.com/keras-team/keras)
repository is dedicated for the development of `keras` code

**Comment by daeing** [comment]
sorry, I mean BatchNormalization. Does different version of tensorflow get different result?

**Comment by daeing** [comment]
I got this from this link(https://github.com/tensorflow/models/blob/master/official/vision/beta/projects/yolo/modeling/layers/nn_blocks.py#:~:text=gamma%2C%20beta%2C%20moving_mean%2C%20moving_variance%20%3D%20self.bn.get_weights()) but I have no idea if the version can cause the different result.

**Comment by tilakrayal** [comment]
@daeing ,
When we call tf.keras.Layers.BatchNormation.get_weights() the order of the list is `[gamma, beta, mean, variance]`.

Also trainable weights does not change. Batchnormalization  maintains non-trainable weights, which are updated via layer updates (i.e. not through backprop): the mean and variance vectors.

If we want to disable weight updates we simply call the layer with the argument training=False in the functional API, which disables the Keras learning phase for layer 

**`x = BatchNormalization()(x, training=False)`**

Batch norm simply shifts and scales the data by a fixed amount derived from the exponential moving averages. This should be fixed at test time and indepdent of the batch contents.

**Comment by tilakrayal** [comment]
Also please test your code in latest stable version 2.8 and let us know if you are facing same issue.Thanks!

**Comment by daeing** [comment]
> @daeing , When we call tf.keras.Layers.BatchNormation.get_weights() the order of the list is `[gamma, beta, mean, variance]`.
> 
> Also trainable weights does not change. Batchnormalization maintains non-trainable weights, which are updated via layer updates (i.e. not through backprop): the mean and variance vectors.
> 
> If we want to disable weight updates we simply call the layer with the argument training=False in the functional API, which disables the Keras learning phase for layer
> 
> **`x = BatchNormalization()(x, training=False)`**
> 
> Batch norm simply shifts and scales the data by a fixed amount derived from the exponential moving averages. This should be fixed at test time and indepdent of the batch contents.

I test my code on tensorflow2.2, the order is [gamma, beta, mean, variance] as you methoned. I know the gamma beta are trainable params, mean and variance are no-trainable params. Because I want bn layer to set_weights() correctly, so I ask this question. Many thanks!!!

**Comment by tilakrayal** [comment]
@daeing ,
Could you please confirm if the issue is resolved. if yes, please feel free to move this issue to closed status
